chain/network/src/peer_manager/peer_manager_actor.rs

```diff
@@ -935,18 +936,33 @@ impl PeerManagerActor {
                 // Sort the shards to keep things tidy
                 shards.sort();
 
+                let peer_id = self.state.config.node_id();
+
+                // Hacky workaround for test environments only.
```

Comment:
To avoid hacky code, you can change the implementation of `is_new` function to also look at the sync hash. Also, for the first epoch with id `1111...11`, there should be no snapshots.

Reply:
Hmm what do you propose exactly? Checking if the sync hash is newer would require adding some dependencies which I would rather avoid it if it's not strictly necessary. Epoch height is convenient because snapshot hosts cache just needs to compare some integers, it doesn't need knowledge of the chain.

Reply:
The issue is that we are using the end of the previous epoch as the sync hash. So the snapshot made of the state at the end of epoch `1111...11` gets assigned height 1, and then the snapshot made at the end of the other epoch with height 1 also gets assigned 1. Wwhen we move the sync hash to the new epoch instead (#12102) this problem will go away.